src/test/java/hudson/plugins/active_directory/ActiveDirectoryDomainTest.java

```diff
@@ -0,0 +1,66 @@
+package hudson.plugins.active_directory;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.FlagRule;
+
+import hudson.util.FormValidation;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+public class ActiveDirectoryDomainTest {
+
+
+	private ActiveDirectorySecurityRealm securityRealm;
+	private AbstractActiveDirectoryAuthenticationProvider authenticationProvider;
+
+	@ClassRule
+	public static FlagRule<String> fipsSystemPropertyRule =
+			FlagRule.systemProperty("jenkins.security.FIPS140.COMPLIANCE", "true");
+
+
+	@Test
+	public void testPasswordTooShortInFIPSMode() {
+		// Create an instance of ActiveDirectoryDomain with a short password and assert exception
+		IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
+			new ActiveDirectoryDomain("example.com", "server", "site", "bindName", "short", TlsConfiguration.TRUST_ALL_CERTIFICATES);
```

Comment:
Beware of using TRUST_ALL_CERTIFICATES. In FIPS mode it won't be allowed after https://github.com/jenkinsci/active-directory-plugin/pull/201 and the test will fail

Reply:
I have updated this in the latest push